gst/nnstreamer/include/tensor_if.h

```diff
@@ -0,0 +1,105 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * GStreamer/NNStreamer Tensor-IF
+ * Copyright (C) 2020 Gichan Jang <gichan2.jang@samsung.com>
+ */
+/**
+ * @file	tensor_if.h
+ * @date	28 Oct 2020
+ * @brief	NNStreamer APIs for tensor_if custom condition
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Gichan Jang <gichan2.jang@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+/**
+ * SECTION:element-tensor_if
+ *
+ * How To for NNdevelopers:
+ *
+ * Define the function in the app.
+ * 1. Define struct, "tensor_if_custom", with the functions defined.
+ * 2. Register the struct with "nnstreamer_if_custom_register" API.
+ * 3. Construct the nnstreamer pipeline and execute it in the app.
+ *
+ * Usage example of the tensor_if custom condition
+ * @code
+ * // Define custom callback function
+ * gboolean tensor_if_custom_cb (const GstTensorsInfo *info,
+ *     const GstTensorMemory * input, gboolean * result) {
+ *   // Describe the conditions and pass the results
+ * }
+ *
+ * void main () {
+ *   GstElement *tif_handle;
+ *   gchar *str_pipeline = g_strdup
+ *     ("videotestsrc pattern=13 ! videoconvert ! videoscale ! video/x-raw, format=RGB, width=640, height=480, framerate=30/1 ! tensor_converter ! mux.sink_0 "
+         videotestsrc pattern=14 ! videoconvert ! videoscale ! video/x-raw, format=RGB, width=640, height=480, framerate=30/1 ! tensor_converter ! mux.sink_1 "
+ *      "tensor_mux name=mux ! tensor_if name=tif compared_value=CUSTOM then=TENSORPICK then-option=0 else=TENSORPICK else-option=1 "
+ *         "tif.src_0 ! queue ! tensor_sink name=sink_true async=false "
+ *         "tif.src_1 ! queue ! tensor_sink name=sink_false async=false");
+ *
+ *   GstElement *pipeline = gst_parse_launch (str_pipeline, NULL);
+ *
+ *   // Get the handle of the tensor_if element
+ *   tif_handle = gst_bin_get_by_name (GST_BIN (pipeline), "tif");
+ *   // Register custom callback function
+ *   nnstreamer_if_custom_register (tif_handle, tensor_if_custom_cb, NULL);
+ *
+ *   // Playing and stop the pipeline or do what you want to do in the app
+ *   gst_element_set_state (pipeline, GST_STATE_PLAYING);
+ *   g_usleep (100000);
+ *   gst_element_set_state (pipeline, GST_STATE_NULL);
+ *   g_usleep (100000);
+ *
+ *   nnstreamer_if_custom_unregister (tif_handle);
+ *   gst_object_unref (tif_handle);
+ *   gst_object_unref (pipeline);
+ *   g_free (str_pipeline);
+ * }
+ * @endcode
```

Comment:
You don't need the full source to run. You just need to show how to use custom IF condition:
```
 * Usage example of the tensor_if custom condition
 * @code
 * // Define custom callback function describing the condition
 * gboolean tensor_if_custom_cb (const GstTensorsInfo *info,
 *     const GstTensorMemory * input, gboolean * result) {
 *   // Describe the conditions and pass the results
 * }
 *
 * ...
 * 
 *   // Get the handle of the tensor_if element
 *   tif_handle = gst_bin_get_by_name (GST_BIN (pipeline), "tif");
 *   // Register custom callback function describing the condition
 *   nnstreamer_if_custom_register (tif_handle, tensor_if_custom_cb, NULL);
 *
 * ...
 *   // Use the condition in a pipeline
 *   // E.g., Pipeline of " ... ! tensor_if name=tif compared_value=CUSTOM then=TENSORPICK then-option=0 else=TENSORPICK else-option=1 ! ... "
 *
 * ...
 *   // After everything is done.
 *   nnstreamer_if_custom_unregister (tif_handle);
 *   gst_object_unref (tif_handle);
 * @endcode
```

Reply:
It'd be more nice if you can do:
```
nnstreamer_if_custom_register ("if_X", tensor_if_custom_cb, NULL);
...
PIPELINE = " ... ! tensor_if compared_value=CUSTOM,if_X ... ! ... "
...
nnstreamer_if_custom_unregister("if_X");
```

Reply:
glib has hash-table. You won't need to reimplement the data structure, too.

Reply:
Changed to register custom callback by the name not element.
for example
```
nnstreamer_if_custom_register ("if_X", tensor_if_custom_cb, NULL);
...
PIPELINE = " ... ! tensor_if compared-value=CUSTOM compared-value-option=if_X ... ! ... "
...
nnstreamer_if_custom_unregister("if_X");
```

Reply:
Uh.. you need to register "tifx" before using the pipeline. :)

Reply:
Update the comment. Thanks.

Reply:
Right! Updated.